src/acl/CharacterSetOption.h

```diff
@@ -33,7 +33,10 @@ inline
 void
 TypedOption<CharacterSetOptionValue>::print(std::ostream &os) const
 {
-    recipient_->value.printChars(os); // TODO: Quote if needed.
+    if (recipient_->valued) {
+        os << " -m=";
+        recipient_->value.printChars(os); // TODO: Quote if needed.
```

Comment:
Even if this change fixes output in one particular use cases, the code does not look right, on several levels (that I can detail if needed). FWIW, the official code is also broken -- I am not defending it.

Here is the default code that we are overwriting with this print() customization:

```C++
    void print(std::ostream &os) const override
    {
        if (configured()) {
            os << ' ' << (disabled() ? offName : onName);
            if (valued())
                os << '=' << recipient_->value;
        }
        // else do not report the implicit default
    }
```

AFAICT, we want to customize the `<< recipient_->value` part (i.e. how the CharacterSet value is printed if it should be printed), not the whole option printing mechanism (that has other useful bits that we do _not_ want to throw away).

If we do not customize at all, then the default CharacterSet printing operator will print the character set name, with decorations:

```C++
std::ostream&
operator <<(std::ostream &s, const CharacterSet &c)
{
    s << "CharacterSet(" << c.name << ')';
    return s;
}
```

I believe the correct solution here is to extract value printing from the above print() into a printValue(std::ostream &) method, so that TypedOption<CharacterSetOptionValue>::printValue() can then customize _just that part_ of print() while leaving the rest of print() alone.

@kinkie, if you can take a stub at this extraction, please do. Otherwise, I will find the time to do it.

Reply:
Sure, I can do this; it feels way overengineered still, but it will work

Reply:
Done.

Here's the output with the current code:
```

acl an2 annotate_transaction n1: n1v3
n2: n2v1

acl an1 annotate_transaction -m=! n1: n1v1
n1: n1v2


```
The newlines are fixed elsewhere